operator/controllers/keptnworkloadinstance/reconcile_predeployment.go

```diff
@@ -0,0 +1,166 @@
+package keptnworkloadinstance
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+
+	klcv1alpha1 "github.com/keptn-sandbox/lifecycle-controller/operator/api/v1alpha1"
+	"github.com/keptn-sandbox/lifecycle-controller/operator/api/v1alpha1/common"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type StatusSummary struct {
+	failed    int
+	succeeded int
+	running   int
+	pending   int
+}
+
+func (r *KeptnWorkloadInstanceReconciler) reconcilePreDeployment(ctx context.Context, req ctrl.Request, workloadInstance *klcv1alpha1.KeptnWorkloadInstance) error {
```

Comment:
this looks very similar between pre- and post-deployment. We might want to unify the similarities into a single function